editor/animation_track_editor.cpp

```diff
@@ -91,6 +91,9 @@ void AnimationTrackKeyEdit::_fix_node_path(Variant &value) {
 
 void AnimationTrackKeyEdit::_update_obj(const Ref<Animation> &p_anim) {
 	if (setting || animation != p_anim) {
+		if (AnimationPlayer* player = AnimationPlayerEditor::get_singleton()->get_player(); player) {
+			player->advance(0.0); // Force an update to refresh the view
+		}
```

Comment:
```suggestion
		AnimationPlayer *player = AnimationPlayerEditor::get_singleton()->get_player();
		if (player) {
			player->advance(0.0); // Force an update to refresh the view.
		}
```
We don't generally allow declarations in if-statements for readability